morphic.js

```diff
@@ -9434,13 +9490,24 @@ MenuItemMorph.prototype.mouseMove = function () {
 };
 
 MenuItemMorph.prototype.mouseClickLeft = function () {
-    if (!this.isListItem()) {
-        this.parent.destroy();
-        this.root().activeMenu = null;
+	if (!(this.doubleClickAction instanceof MenuMorph)) {
+        if (!this.isListItem()) {
+            var topmenu = this.parent;
+            while (topmenu.parent instanceof MenuMorph) {
+                topmenu = topmenu.parent;
+            }
+            topmenu.destroy();
+            this.root().activeMenu = null;
+        }
+        this.trigger();
     }
-    this.trigger();
 };
 
+MenuItemMorph.prototype.mouseDoubleClick = function () {
+    if (!(this.doubleClickAction instanceof MenuMorph))
```

Comment:
Fine.  I do always do that, but didn't in this case because the IF is the entire body of the procedure!  But okay, just for you.